force-app/main/default/classes/ContactTriggerHandler.cls

```diff
@@ -0,0 +1,112 @@
+/**
+ * File:        ContractTriggerHandler.trigger
+ * Project:     OSFPracticalExercises
+ * Date:        August 18, 2021
+ * Created By:  Can Yavuzkurt
+ * *************************************************************************
+ * Description: Handles Contact Trigger
+ * *************************************************************************
+ * History:
+ * Date:                Modified By:                         Description:
+ * August 18, 2021      Can Yavuzkurt                        Consolidated PhoneTrigger and ValidateTrigger here
+ */
+public with sharing class ContactTriggerHandler {
+    
+    /**
+     * @author Can Yavuzkurt
+     * @date August 18, 2021
+     * @description Validates contact info before insert/update
+     * @name handleBefore
+     * @param contacts List<Contact> Trigger.new supplied by the Trigger
+    */
+    public static void handleBefore(List<Contact> contacts) {
+
+        // Get AccountIds of new contacts for query reasons
+        List<Id> accountIds = new List<Id>();
+        for (Contact contact : contacts) {
+            
+            accountIds.add(contact.AccountId);
+        }
+
+        // Construct AccountId, PrimaryContact pair map
+        Map<Id, Contact> primContactMap = new Map<Id, Contact>();
+        try {
+            
+            List<Contact> primContacts = [SELECT Id, AccountId FROM Contact WHERE AccountId IN :accountIds AND Is_Primary_Contact__c = true];
+        } catch (QueryException e) {
+            
+            throw new HandledException(System.Label.Query_Error);
+        }
+        for (Contact primContact : primContacts) {
+            
+            primContactMap.put(primContact.AccountId, primContact);
+        }
+
+        // Validate 
+        for (Contact contact : contacts) {
+            
+            Contact oldPrimaryContact = primContactMap.get(contact.AccountId);
+            if ( contact.Is_Primary_Contact__c && oldPrimaryContact != null && oldPrimaryContact.Id != contact.Id) {
+
+                contact.addError('There is a primary contact associated with the account already.');
```

Comment:
All strings have to be moved to custom labels OR constants (public static final String UPPER_CASE_NAME = 'string';)

Reply:
done